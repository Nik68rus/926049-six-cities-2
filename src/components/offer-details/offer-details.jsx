import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {makeFirstCharCapital, getStatus, getUserAvatar} from '../../util';
import {OfferType, CardType} from '../../constants';
import ReviewList from '../review-list/review-list';
import Map from '../map/map';
import OffersList from '../offers-list/offers-list';
import withActiveCard from '../../hocs/with-active-card';
import {Operation} from '../../store/action/action-creator';

const OffersListWrapped = withActiveCard(OffersList);

export const OfferDetails = (props) => {
  const {offer, isFavorite, offers, user, isAuthorizationRequired, city, activePin, reviews, onReviewSubmit, onBookmarkClickHandler, onFavoriteClickHandler} = props;
  const {id, title, price, rate, isPremium, photos, type, bedrooms, description, host, goods, maxAdults} = offer;

  const getNeighborOffers = (qtty) => {
    return offers.filter((cityOffer) => cityOffer.id !== offer.id).slice(0, qtty);
  };

  const cardMark = (mark) => {
    return mark ? <div className="property__mark">
      <span>Premium</span>
    </div> : ``;
  };

  const neighborOffers = getNeighborOffers(3);

  const bookmarkCard = (bookmark) => {
    return bookmark ? `property__bookmark-button property__bookmark-button--active button` : `property__bookmark-button button`;
  };

  return <div className="page">
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link">
              <img className="header__logo" src="../img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link
                  className="header__nav-link header__nav-link--profile"
                  to={isAuthorizationRequired ? `/login` : `/favorites`}
                  onClick={isAuthorizationRequired ? null : onFavoriteClickHandler}
                >
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                    {getUserAvatar(isAuthorizationRequired, user)}
                  </div>
                  <span className="header__user-name user__name">{isAuthorizationRequired ? `Sign In` : user.email}</span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>

    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {
              photos.map((photo, i) => <div key={`id${id}-photo${i}`} className="property__image-wrapper">
                <img className="property__image" src={photo} alt={title} />
              </div>)
            }
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            {cardMark(isPremium)}
            <div className="property__name-wrapper">
              <h1 className="property__name">
                {title}
              </h1>
              <button className={bookmarkCard(isFavorite)} type="button" onClick={() => onBookmarkClickHandler(id, getStatus(isFavorite))}>
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: `${rate * 20}%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{rate}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {makeFirstCharCapital(OfferType[type.toUpperCase()])}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                Max {maxAdults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">&euro;{price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">
                {
                  goods.map((good) => <li key={good} className="property__inside-item">
                    {good}
                  </li>)
                }
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={`../${host.avatar}`} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {host.name}
                </span>
                <span className="property__user-status">
                  {host.isPro ? `Pro` : ``}
                </span>
              </div>
              <div className="property__description">
                <p className="property__text">
                  {description}
                </p>
              </div>
            </div>
            <ReviewList reviews={reviews} id={offer.id} onReviewSubmit={onReviewSubmit} isAuthorizationRequired={isAuthorizationRequired}/>
          </div>
        </div>
        <section className="property__map map">
          <Map
            city={city}
            offers={[offer, ...neighborOffers]}
            activePin={activePin}
          />
        </section>
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <OffersListWrapped cardType={CardType.NEAR} offers={neighborOffers} />
        </section>
      </div>
    </main>
  </div>;
};

OfferDetails.propTypes = {
  offers: PropTypes.array.isRequired,
  offer: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rate: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isBookmarked: PropTypes.bool.isRequired,
    photos: PropTypes.array.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    goods: PropTypes.array.isRequired,
    maxAdults: PropTypes.number.isRequired,
    host: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      isPro: PropTypes.bool.isRequired,
      avatar: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
  }).isRequired,
  city: PropTypes.shape({}).isRequired,
  activePin: PropTypes.number.isRequired,
  reviews: PropTypes.array.isRequired,
  onReviewSubmit: PropTypes.func.isRequired,
  onBookmarkClickHandler: PropTypes.func.isRequired,
  onFavoriteClickHandler: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isAuthorizationRequired: state.user.isAuthorizationRequired,
  user: state.user.user,
  city: state.user.city,
  offers: state.user.cityOffers,
  activePin: state.user.activePinID,
  reviews: state.data.reviews,
  isFavorite: state.data.allOffers.find((it) => {
    return it.id === ownProps.offer.id;
  }).isBookmarked,
});

const mapDispatchToProps = (dispatch) => ({
  onReviewSubmit: (id, review) => {
    dispatch(Operation.postReview(id, review));
  },
  onBookmarkClickHandler: (id, status) => dispatch(Operation.changeOfferStatus(id, status)),
});

export default connect(mapStateToProps, mapDispatchToProps)(OfferDetails);
