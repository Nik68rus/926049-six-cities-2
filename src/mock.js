export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Ivan`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 3,
    comment: `Beautiful place in beautifool location with good neighbors!`,
    date: Date.now(),
  },
  {
    id: 2,
    user: {
      id: 6,
      isPro: false,
      name: `Irina`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 5,
    comment: `It was awesome!`,
    date: Date.now(),
  },
  {
    id: 3,
    user: {
      id: 7,
      isPro: true,
      name: `Max`,
      avatar: `../img/avatar-max.jpg`,
    },
    rating: 2,
    comment: `I would never recomend that!`,
    date: Date.now(),
  },

];
