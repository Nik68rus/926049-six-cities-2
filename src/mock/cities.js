export const CITIES = [
  {
    name: `Amsterdam`,
    location: {
      latitude: 52.38333,
      longtitude: 4.9,
      zoom: 12,
    }
  },
  {
    name: `Paris`,
    location: {
      latitude: 48.8544,
      longtitude: 2.348868,
      zoom: 10,
    }
  },
  {
    name: `Cologne`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Brussels`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Hamburg`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 52.370216,
      longtitude: 4.895168,
      zoom: 10,
    }
  },
];
