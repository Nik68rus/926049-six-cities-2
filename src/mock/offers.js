import {Type} from '../constants';
import {CITIES} from './cities';

export const getOffers = () => {
  let allOffers = [];
  const offersByCity = CITIES.map((city) => {
    return [
      {
        id: 0,
        city,
        title: `Beautiful & luxurious apartment at great location in ${city.name}`,
        coords: [52.3909553943508, 4.85309666406198],
        picture: `img/apartment-01.jpg`,
        type: Type.APARTMENT,
        price: 150,
        rate: 50,
        isBookmarked: true,
        isPremium: false,
      },
      {
        id: 1,
        city,
        title: `Wood and stone place in ${city.name}`,
        coords: [52.369553943508, 4.85309666406198],
        picture: `img/room.jpg`,
        type: Type.ROOM,
        price: 80,
        rate: 80,
        isBookmarked: false,
        isPremium: false,
      },
      {
        id: 2,
        city,
        title: `Canal View Prinsengracht in ${city.name}`,
        coords: [52.3909553943508, 4.929309666406198],
        picture: `img/apartment-03.jpg`,
        type: Type.APARTMENT,
        price: 120,
        rate: 90,
        isBookmarked: false,
        isPremium: false,
      },
      {
        id: 3,
        city,
        title: `Nice, cozy, warm big bed apartment in ${city.name}`,
        coords: [52.3809553943508, 4.939309666406198],
        picture: `img/apartment-02.jpg`,
        type: Type.APARTMENT,
        price: 200,
        rate: 100,
        isBookmarked: true,
        isPremium: true,
      },
    ];
  });
  offersByCity.forEach((cityOffers) => {
    cityOffers.forEach((offer) => allOffers.push(offer));
  });
  allOffers.forEach((offer, i) => {
    offer.id = i;
  });
  return allOffers;
};

// export const offers = getOffers();

export const offers = [
  {
    id: 0,
    city: CITIES[0],
    title: `Beautiful & luxurious apartment at great location in Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 1,
    city: CITIES[0],
    title: `Wood and stone place in Amsterdam`,
    coords: [52.369553943508, 4.85309666406198],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 2,
    city: CITIES[0],
    title: `Canal View Prinsengracht in Amsterdam`,
    coords: [52.3909553943508, 4.929309666406198],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 3,
    city: CITIES[0],
    title: `Nice, cozy, warm big bed apartment in Amsterdam`,
    coords: [52.3809553943508, 4.939309666406198],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 4,
    city: CITIES[1],
    title: `Beautiful & luxurious apartment at great location in Paris`,
    coords: [48.866714, 2.338152],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 5,
    city: CITIES[1],
    title: `Wood and stone place in Paris`,
    coords: [48.858179, 2.365554],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 6,
    city: CITIES[1],
    title: `Canal View Prinsengracht in Paris`,
    coords: [48.840967, 2.345730],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 7,
    city: CITIES[1],
    title: `Nice, cozy, warm big bed apartment in Paris`,
    coords: [48.852996, 2.324593],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 8,
    city: CITIES[2],
    title: `Beautiful & luxurious apartment at great location in Cologne`,
    coords: [50.930207, 6.923582],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 9,
    city: CITIES[2],
    title: `Wood and stone place in Cologne`,
    coords: [50.934906, 6.972391],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 10,
    city: CITIES[2],
    title: `Canal View Prinsengracht in Cologne`,
    coords: [50.948242, 6.951473],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 11,
    city: CITIES[3],
    title: `Nice, cozy, warm big bed apartment in Brussels`,
    coords: [50.839652, 4.334081],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 12,
    city: CITIES[3],
    title: `Beautiful & luxurious apartment at great location in Brussels`,
    coords: [50.865504, 4.368736],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 13,
    city: CITIES[4],
    title: `Wood and stone place in Hamburg`,
    coords: [53.585260, 9.984106],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 14,
    city: CITIES[4],
    title: `Canal View Prinsengracht in Hamburg`,
    coords: [53.558643, 9.967173],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 15,
    city: CITIES[4],
    title: `Nice, cozy, warm big bed apartment in Hamburg`,
    coords: [53.542315, 9.994189],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
  {
    id: 16,
    city: CITIES[4],
    title: `Beautiful & luxurious apartment at great location in Hamburg`,
    coords: [53.542109, 9.965807],
    picture: `img/apartment-01.jpg`,
    type: Type.APARTMENT,
    price: 150,
    rate: 50,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 17,
    city: CITIES[4],
    title: `Wood and stone place in Hamburg`,
    coords: [53.550431, 9.976893],
    picture: `img/room.jpg`,
    type: Type.ROOM,
    price: 80,
    rate: 80,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 18,
    city: CITIES[4],
    title: `Canal View Prinsengracht in Hamburg`,
    coords: [53.545377, 9.992105],
    picture: `img/apartment-03.jpg`,
    type: Type.APARTMENT,
    price: 120,
    rate: 90,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 19,
    city: CITIES[5],
    title: `Nice, cozy, warm big bed apartment in Dusseldorf`,
    coords: [51.235248, 6.790136],
    picture: `img/apartment-02.jpg`,
    type: Type.APARTMENT,
    price: 200,
    rate: 100,
    isBookmarked: true,
    isPremium: true,
  },
];
