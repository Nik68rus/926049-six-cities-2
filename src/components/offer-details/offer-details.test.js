import renderer from 'react-test-renderer';
import {OfferDetails} from './offer-details';
import {offer, user, city, review} from '../../mock';

jest.mock(`../review-list/review-list`, () => `ReviewList`);
jest.mock(`../offers-list/offers-list`, () => `OffersList`);
jest.mock(`../map/map`, () => `Map`);

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`Offer details correctly renders after relaunch`, () => {
  const offerDetails = renderer
    .create(<OfferDetails
      offers={[offer]}
      offer={offer}
      isFavorite={false}
      isAuthorizationRequired={false}
      user={user}
      city={city}
      activePin={0}
      reviews={[review]}
      onReviewSubmit={jest.fn()}
      onBookmarkClickHandler={jest.fn()}
      onFavoriteClickHandler={jest.fn()}
    />)
  .toJSON();

  expect(offerDetails).toMatchSnapshot();
});
