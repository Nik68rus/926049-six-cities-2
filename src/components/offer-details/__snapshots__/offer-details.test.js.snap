// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer details correctly renders after relaunch 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        />
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            />
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious studio at great location"
              className="property__image"
              src="img/1.png"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious studio at great location"
              className="property__image"
              src="img/2.png"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious studio at great location
            </h1>
            <button
              className="property__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "96%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Cable TV
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="../img/1.png"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              </p>
            </div>
          </div>
          <ReviewList
            id={0}
            isAuthorizationRequired={false}
            onReviewSubmit={[MockFunction]}
            reviews={
              Array [
                Object {
                  "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "date": "2019-05-08T14:13:56.569Z",
                  "id": 1,
                  "rating": 4,
                  "user": Object {
                    "avatar": "img/1.png",
                    "id": 4,
                    "isPro": false,
                    "name": "Max",
                  },
                },
              ]
            }
          />
        </div>
      </div>
      <section
        className="property__map map"
      >
        <Map
          activePin={0}
          city={
            Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Amsterdam",
            }
          }
          offers={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "coords": Array [
                  52.35514938496378,
                  4.673877537499948,
                ],
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar": "img/1.png",
                  "id": 0,
                  "isPro": false,
                  "name": "Angelina",
                },
                "id": 0,
                "isBookmarked": false,
                "isPremium": false,
                "locationZoom": 8,
                "maxAdults": 4,
                "photos": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "picture": "img/1.png",
                "price": 120,
                "rate": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OffersList
          activeCard={-1}
          cardType="NEAR"
          mouseEnterHandler={[Function]}
          offers={Array []}
        />
      </section>
    </div>
  </main>
</div>
`;
